Add ability to support multiple feed services

jtapia opened the pull request:
Quick Info
---
Migrations? :+1:

What does this change?
----
- Add support for `Google Feed`, `Google Merchant Center` and `Facebook Feed`
- Add admin `product_feed` and `product_catalog` controller and test case
- Add spree `product_feed` and `product_catalog` model and test case
- Add spree `product_feed` and `product_catalog` table[migration]
- Add `product_feed` and `product_catalog` views
- Fix `product_feed` and `product_catalog` styles
- Add logic to handle the type of the response(xml or csv)

How do you manually test this?
- Go to `admin` section
- Click on `navbar` icon or go to `/admin/product_feeds`
----

Screenshots
----
http://recordit.co/y8jD1VjX8i

fastjames commented:
From watching the screencast, it looks like the store owner gets to choose one format and that determines what `/products.rss` looks like. Is that correct?

jtapia commented:
@fastjames I found out that facebook supports the xml schema, so I changed the logic to not set a format, I just updated the screencast and also now is pointing to `gemini_master `, thanks

fastjames commented:
[ch1416]

joshcampbell commented on .gitignore:
```diff
@@ -10,3 +10,4 @@ nbproject
 spec/dummy
 pkg
 Gemfile.lock
+coverage/
```
What tool creates this?

fastjames replied:
I think that's `simplecov`.

jtapia commented on app/controllers/spree/products_controller_decorator.rb:
```diff
@@ -1,18 +1,44 @@
-  class << self
-    def prepended(klass)
-      klass.respond_to :rss, :xml, only: :index
+Spree::ProductsController.class_eval do
```
Not really, I consider that we don't need to define a `respond_to` method, instead we can use a `before_action`, what do you think?

fastjames commented:
Merged.

aneekam commented:
https://app.clubhouse.io/enginecommerce/story/1138/product-feeds-should-include-variants-rather-than-products
